### ISSUE TITLE
Riemann Zeta function implementation template

### ISSUE THREAD

**Issue by Bobingstern** [josdejong/mathjs#2904]
I've coded up an implementation to calculate the riemann zeta function over the entire complex plane using math.js. It works very quickly for real numbers and uses a number of terms proportional to $\approx (1.3*\ln(d * 0.1)+ 0.9*|t|)$ where $d$ is the number of decimal places of accuracy and $t$ is the imaginary part of the input. I've implemented this in my small js library called `special.js`. Here is a description of the algorithm used:

Computation makes use of the function $\displaystyle d_k = n\sum_{j=k}^n \frac{(n+j-1)!4^j}{(n-j)!(2j)!}$ where $n$ is chosen arbitrarily making use of the precision and uses $\displaystyle\zeta(s) = \frac{1}{d_0(1-2^{1-s})}\sum_{k=1}^n \frac{(-1)^{k-1}d_k}{k^s}$ for $\Re(s) > 1$ and uses the functional equation $\displaystyle \zeta(s) = 2^s\pi^{s-1}\sin\bigg(\frac{\pi s}{2}\bigg)\Gamma(1-s)\zeta(1-s)$ where $\Gamma(s)$ is the gamma function to extend it to the entire complex plane.

The js code in math.js is as follows:
```js
const math = require("mathjs")

function zeta(s, prec, only){
	s = math.complex(s)
    if (s.re == 0 && s.im == 0){
        return -0.5
    }
    prec = prec == undefined ? 1e-3 : prec
    let dec = -Math.round(Math.log(prec*0.1)/Math.log(10))
    let n = Math.round(1.3*dec + 0.9*Math.abs(s.im))
    n = Math.min(n, 60)
    
    function d(k){
        let S = 0
        for (let j=k;j<=n;j++){
            S += (math.factorial(n+j-1) * (4**j) / (math.factorial(n-j)*math.factorial(2*j)  ))
        }
        return n*S
    }
    function f(s){
        let c = math.divide(1, 
            math.multiply(d(0), math.subtract(1, math.pow(2, math.subtract(1, s)))))
        let S = math.complex(0, 0)
        for (let k=1;k<=n;k++){
            S = S.add(
                math.divide((-1)**(k-1) * d(k), math.pow(k, s))
            )
        }
        return math.multiply(c, S)
    }
    if (s.re > 1 || s.re == 0 || only){
        //console.log("Running "+ n +" iterations")
        return f(s)
    }
    else{
        let c = math.multiply(math.pow(2, s), math.pow(Math.PI, math.subtract(s, 1)))
        c = math.multiply(c, (math.sin(math.multiply(Math.PI/2, s))))
        c = math.multiply(c, math.gamma(math.subtract(1, s)))
        //Recursion
        return math.multiply(c, zeta(math.subtract(1, s), prec, true))
    }
}

function benchmark(s, n){
	let avg = 0
	for (let i=0;i<n;i++){
		let now = performance.now()
		zeta(s, 0.0000001)
		avg += performance.now() - now
	}
	return avg/n
}
console.log(benchmark(math.complex(1/2, 14.134725), 100)) //First zero of the zeta function
```

I'm not very experienced in node.js libraries and such so I hope that someone more experienced can add it to the library.

**Comment by josdejong** [comment]
Thanks for sharing Anik! Would be nice indeed to implement the Riemann Zeta function in mathjs itself.

Anyone able to work out the function in a PR and add tests and documentation? Help would be welcome.

**Comment by Bobingstern** [comment]
![image](https://user-images.githubusercontent.com/74193405/221024422-ca70455f-e239-4e03-b364-a43d3fbba6a7.png)

I've added a new enhancement using the Riemann Siegel formula to evaluate on the critical line much faster. The code itself is very ugly since the remainder term calculation is horrible and so I used a taylor series for the derivatives and such. The graph is that of the zeta function on the critical line with varying imaginary inputs with blue and orange representing the real and imaginary output respectively. You can read about the Riemann Siegel Z(t) function [here](https://axion004.files.wordpress.com/2022/12/calculating_zeros_of_the_riemann_zeta_function.pdf). The new code is slightly changed and is as follows. I'll try my best to figure out how to incorporate this into math.js and try to send a PR with tests and docs.
```js
const math = require("mathjs")

function zeta(s, prec, only){
	s = math.complex(s)
    if (s.re == 0 && s.im == 0){
        return -0.5
    }
    prec = prec == undefined ? 1e-3 : prec
    let dec = -Math.round(Math.log(prec*0.1)/Math.log(10))
    let n = Math.round(1.3*dec + 0.9*Math.abs(s.im))
    //n = Math.min(n, 60)
    if (s.re == 0.5 && s.im >= 1){
        return math.divide(RiemannZ(s.im, 5), math.exp(math.complex(0, theta(s.im))))
    }
    function d(k){
        let S = 0
        for (let j=k;j<=n;j++){
            S += (math.factorial(n+j-1) * (4**j) / (math.factorial(n-j)*math.factorial(2*j)  ))
        }
        return n*S
    }
    function f(s){
        let c = math.divide(1, 
            math.multiply(d(0), math.subtract(1, math.pow(2, math.subtract(1, s)))))
        let S = math.complex(0, 0)
        for (let k=1;k<=n;k++){
            S = S.add(
                math.divide((-1)**(k-1) * d(k), math.pow(k, s))
            )
        }
        return math.multiply(c, S)
    }
    if (s.re > 1 || s.re == 0 || only){
        //console.log("Running "+ n +" iterations")
        return f(s)
    }
    else{
        let c = math.multiply(math.pow(2, s), math.pow(Math.PI, math.subtract(s, 1)))
        c = math.multiply(c, (math.sin(math.multiply(Math.PI/2, s))))
        c = math.multiply(c, math.gamma(math.subtract(1, s)))
        //Recursion
        return math.multiply(c, zeta(math.subtract(1, s), prec, true))
    }
}

//Hellish coefficents function. It is minifed since its just a bunch of return statements
function C($,_){return 0==$?.3826834323650898*_**0+.43724046807752043*_**2+.1323765754803435*_**4-.013605026047674188*_**6-.013567621970103581*_**8-.0016237253231444653*_**10+2970535373337969e-19*_**12+794330087952147e-19*_**14+46556124614505e-20*_**16-143272516309551e-20*_**18-10354847112313e-20*_**20+1235792708386e-20*_**22+17881083858e-19*_**24-339141439e-19*_**26-163266339e-19*_**28-37851093e-20*_**30+9327423e-20*_**32+522184e-20*_**34-33507e-20*_**36-3412e-20*_**38+58e-20*_**40+15e-20*_**42:1==$?-.026825102628375348*_**1+.013784773426351853*_**3+.03849125048223508*_**5+.009871066299062077*_**7-.0033107597608584044*_**9-.0014647808577954152*_**11-1320794062487696e-20*_**13+5922748701847141e-20*_**15+598024258537345e-20*_**17-96413224561698e-20*_**19-18334733722714e-20*_**21+446708756272e-20*_**23+270963508218e-20*_**25+7785288654e-20*_**27-2343762601e-20*_**29-158301728e-20*_**31+12119942e-20*_**33+1458378e-20*_**35-28786e-20*_**37-8663e-20*_**39-84e-20*_**41+36e-20*_**43+1e-20*_**45:2==$?.005188542830293168*_**0+30946583880634744e-20*_**2-.011335941078229373*_**4+.0022330457419581446*_**6+.00519663740886233*_**8+3439914407620834e-19*_**10-5910648427470583e-19*_**12-10229972547935857e-20*_**14+2088839221699276e-20*_**16+592766549309654e-20*_**18-16423838362436e-20*_**20-15161199700941e-20*_**22-590780369821e-20*_**24+209115148595e-20*_**26+17815649583e-20*_**28-1616407246e-20*_**30-238069625e-20*_**32+5398265e-20*_**34+1975014e-20*_**36+23333e-20*_**38-11188e-20*_**40-416e-20*_**42+44e-20*_**44+3e-20*_**46:3==$?-.0013397160907194568*_**1+.003744215136379394*_**3-.0013303178919321468*_**5-.0022654660765471786*_**7+9548499998506731e-19*_**9+6010038458963604e-19*_**11-10128858286776622e-20*_**13-6865733449299826e-20*_**15+59853667915386e-20*_**17+333165985123995e-20*_**19+21919289102435e-20*_**21-7890884245681e-20*_**23-94146850813e-19*_**25+95701162109e-20*_**27+18763137453e-20*_**29-443783768e-20*_**31-224267385e-20*_**33-3627687e-20*_**35+1763981e-20*_**37+79608e-20*_**39-942e-19*_**41-713e-20*_**43+33e-20*_**45+4e-20*_**47:46483389361763383e-20*_**0-.001005660736534047*_**2+24044856573725794e-20*_**4+.0010283086149702322*_**6-7657861071755644e-19*_**8-20365286803084818e-20*_**10+2321229049106873e-19*_**12+326021442438652e-19*_**14-2557906251794953e-20*_**16-410746443891574e-20*_**18+117811136403713e-20*_**20+24456561422485e-20*_**22-2391582476734e-20*_**24-750521420704e-20*_**26+13312279416e-20*_**28+13440626754e-20*_**30+351377004e-20*_**32-151915445e-20*_**34-8915418e-20*_**36+1119589e-20*_**38+10516e-19*_**40-5179e-20*_**42-807e-20*_**44+11e-20*_**46+4e-20*_**48}

function theta(t){
    twopi = 2 * math.pi
    return (t/2.0 * math.log(t/twopi) - t/2.0 - math.pi/8.0
            + 1.0/(48.0*t) + 7.0/(5760*t**3))
}
//Riemann Z(t) formula
function RiemannZ(t, r){
    tDivTwoPi = t / (2 * math.pi)
    val = math.sqrt(tDivTwoPi)
    N = math.floor(math.abs(val))
    SUM = 0

    i = 1
    while (i<=N){
        SUM += (math.cos(theta(t) - t * math.log(i))) / math.sqrt(i)
        i += 1
    }
    SUM *= 2
    frac = val - N
    k = 0
    R = 0

    //Remainder Calculation
    while (k <= r){
        R += C(k, 2.0*frac-1.0) * tDivTwoPi**(k * (-0.5))
        k += 1
    }
    remainder = (-1)**(N-1) * tDivTwoPi**(-0.25) * R
    return SUM + remainder
}
function benchmark(s, n){
	let avg = 0
	for (let i=0;i<n;i++){
		let now = performance.now()
		zeta(s, 0.0000001)
		avg += performance.now() - now
	}
	return avg/n
}

console.log(benchmark(math.complex(1, 167.235364), 100)) //Normal zeta function testing

console.log(benchmark(math.complex(1/2, 167.235364), 100)) //Riemann Siegel Critical zeta function testing (Triggered if Re(s) is 1/2). Much Much faster
```

**Comment by HanchaiN** [comment]
Maybe we can make the hellish coefficients function more readable by storing the whole coefficient like in [erf.js](https://github.com/josdejong/mathjs/blob/7950d9c1f319242a41f8716eb6cbe352f2ce6384/src/function/special/erf.js#LL66C8-L66C8). This may cost more space and time since it will have to either store unused coefficient (0s) or figures out if you will use odd or even coefficients. The latter might be better in terms of performance though.
Here's an example (the coefficient is not included but will be referred to as `c`):
```javascript
function C(n, x){
    const x2 = x * x;
    const m = n < 4 ? n : 4;
    const y = c[m].reduceRight((acc, v) => acc * x2 + v, 0);
    if (m % 2 === 1) return x * y;
    return y;
}
```
It should be noted that `RiemannZ(t, r)` seems to be called only with `RiemannZ(s.im, 5)` so the while loop in the remainder calculation (`while (k <= r) {...}`) might as well be replaced as a for-loop. Also, the function `C` only provides cases for `n` from `1` to `4` or more while the function will receive `n` from `0` to `5`.

**Comment by Bobingstern** [comment]
I'm thinking about replacing the coefficients function with the actual exact function since currently it is a Taylor series and will not be accurate everywhere. The actual expressions is, unsurprisingly, extremely complicated and requires high order derivatives of Psi(x) defined by cos(2pi(x^2-x-1/16))/cos(2pi x). It can be calculated beforehand but the expression is massive. I'll try to figure a better way to calculate them

**Comment by Bobingstern** [comment]
Here is a new version that uses an array of coefficients that I recalculated. It should be *slightly* more accurate. I also increased the size of the theta function asymptotic series. If this ever becomes a part of the library make the documentation says that on the critical strip Re(s)=1/2 it is more accurate for larger imaginary parts of s.
```js
const math = require("mathjs")

function zeta(s, prec, only){
	s = math.complex(s)
    if (s.re == 0 && s.im == 0){
        return -0.5
    }
    prec = prec == undefined ? 1e-3 : prec
    let dec = -Math.round(Math.log(prec*0.1)/Math.log(10))
    let n = Math.round(1.3*dec + 0.9*Math.abs(s.im))
    //n = Math.min(n, 60)
    if (s.re == 0.5 && s.im >= 1){
        return math.divide(RiemannZ(s.im, 5), math.exp(math.complex(0, theta(s.im))))
    }
    function d(k){
        let S = 0
        for (let j=k;j<=n;j++){
            S += (math.factorial(n+j-1) * (4**j) / (math.factorial(n-j)*math.factorial(2*j)  ))
        }
        return n*S
    }
    function f(s){
        let c = math.divide(1, 
            math.multiply(d(0), math.subtract(1, math.pow(2, math.subtract(1, s)))))
        let S = math.complex(0, 0)
        for (let k=1;k<=n;k++){
            S = S.add(
                math.divide((-1)**(k-1) * d(k), math.pow(k, s))
            )
        }
        return math.multiply(c, S)
    }
    if (s.re > 1 || s.re == 0 || only){
        //console.log("Running "+ n +" iterations")
        return f(s)
    }
    else{
        let c = math.multiply(math.pow(2, s), math.pow(Math.PI, math.subtract(s, 1)))
        c = math.multiply(c, (math.sin(math.multiply(Math.PI/2, s))))
        c = math.multiply(c, math.gamma(math.subtract(1, s)))
        //Recursion
        return math.multiply(c, zeta(math.subtract(1, s), prec, true))
    }
}

//Recalculated Taylor Coefficients of the C_n remainder terms

let COEFFS = [
    [0.923879534973963, -2.40447104083258, 2.40447377585498, 4.83169016036145, -18.2361922325210, 27.7643978409119, -15.7966469795015, -20.4631454539165, 60.5433282970305, -70.9377645282682, 37.2985079016296, 16.7905276267356, -43.8106258641246, 9.17771014597747, 80.0739039991204, -179.996536374092, 240.689321443439, -238.442749679089, 185.468254983425, -114.747143268585, 55.8878343105316, -20.7833318710327, 5.56842517852783, -0.959747314453125, 0.0799789428710938],
    [-0.0305970353775537, 0.461928145532058, -1.75820103272372, 2.00066871192907, 4.53545338407905, -21.4701258964562, 37.7343659014643, -28.3434672651465, -17.5439976228023, 61.0354864441400, -16.6218991352115, -184.574907154224, 518.627828950598, -853.542442855211, 1026.77031920780, -958.385019272977, 704.118785494245, -403.461783552932, 175.043942145129, -54.3041951483027, 10.7635551083289, -1.02510048650751], 
    [0.00127856271992947, -0.0115466350873615, 0.333359693594379, -1.51045611934403, 2.39077353971839, 1.75012654345750, -10.8659613189895, -1.77037119495183, 101.743226064846, -358.343400702722, 763.776931762499, -1181.11539619777, 1403.77938104144, -1308.32927606516, 960.134517130875, -554.936346813697, 256.461039430787, -100.979903253844, 37.9386791985220, -13.8192585904049, 4.07281463612270, -0.768825364880632, 0.0698932149891484], 
    [-5.65130769148159e-5, -0.00372776078640970, 0.0492496949365606, -0.286875287820373, 2.18109875136261, -14.2639964275846, 61.7952336905414, -185.421279606555, 409.383383115748, -693.204245855566, 926.834642606634, -1003.90223210110, 907.532033548975, -708.523774479464, 488.094492238016, -291.511090813811, 142.095108962924, -52.3287250677516, 13.8120360398578, -2.86014536988476, 0.690962929520245, -0.193943554101081, 0.0349466074945742, -0.00303883543431080], 
    [0.000908082635560355, -0.0224412125527388, 0.268504907549843, -1.94414079977184, 9.94514567495564, -37.2973724955692, 105.554663705919, -232.745142224515, 412.554002080475, -603.471125537491, 739.245143701107, -757.270719521628, 639.413356553839, -438.878154310469, 247.439533777360, -121.667630160839, 56.3694016604538, -23.8824770337663, 7.90572425811156, -1.70965456010061, 0.206121626994396, -0.0164514983219106, 0.00440780804775184, -0.000759708858577700, 6.33090715481417e-5], 
    [0.00267275793846456, -0.0571237930695642, 0.556216298035879, -3.39475759249389, 14.6207281417306, -47.1719852983861, 118.105321729659, -234.363035897620, 373.390653322949, -483.485507704209, 519.154267111751, -476.737302427857, 383.932700487246, -268.254714952686, 154.492728440213, -69.4944471932793, 24.6060256149920, -7.97076896340324, 2.77119678243209, -0.859713332682895, 0.170402522464908, -0.0162288116633246], 
    [0.00492368137865323, -0.0909698000538868, 0.797172121382113, -4.38808589905504, 16.9662997501199, -48.9377038546898, 109.905279292975, -199.069777821749, 298.210787383758, -372.455465435712, 384.673710418843, -325.772733913638, 230.074513827627, -143.108873617560, 81.9200997250781, -41.0408508895021, 15.9430481290959, -4.36958924216078, 0.908505117500893, -0.218778693349901, 0.0644784999512171, -0.0121716087474934, 0.00110650988613577]
]


function C(n, x){
    const m = n <= 5 ? n : 5;
    const y = COEFFS[m].reduceRight((acc, v) => acc * x + v, 0);
    return y;
}

//function C($,_){return 0==$?.3826834323650898*_**0+.43724046807752043*_**2+.1323765754803435*_**4-.013605026047674188*_**6-.013567621970103581*_**8-.0016237253231444653*_**10+2970535373337969e-19*_**12+794330087952147e-19*_**14+46556124614505e-20*_**16-143272516309551e-20*_**18-10354847112313e-20*_**20+1235792708386e-20*_**22+17881083858e-19*_**24-339141439e-19*_**26-163266339e-19*_**28-37851093e-20*_**30+9327423e-20*_**32+522184e-20*_**34-33507e-20*_**36-3412e-20*_**38+58e-20*_**40+15e-20*_**42:1==$?-.026825102628375348*_**1+.013784773426351853*_**3+.03849125048223508*_**5+.009871066299062077*_**7-.0033107597608584044*_**9-.0014647808577954152*_**11-1320794062487696e-20*_**13+5922748701847141e-20*_**15+598024258537345e-20*_**17-96413224561698e-20*_**19-18334733722714e-20*_**21+446708756272e-20*_**23+270963508218e-20*_**25+7785288654e-20*_**27-2343762601e-20*_**29-158301728e-20*_**31+12119942e-20*_**33+1458378e-20*_**35-28786e-20*_**37-8663e-20*_**39-84e-20*_**41+36e-20*_**43+1e-20*_**45:2==$?.005188542830293168*_**0+30946583880634744e-20*_**2-.011335941078229373*_**4+.0022330457419581446*_**6+.00519663740886233*_**8+3439914407620834e-19*_**10-5910648427470583e-19*_**12-10229972547935857e-20*_**14+2088839221699276e-20*_**16+592766549309654e-20*_**18-16423838362436e-20*_**20-15161199700941e-20*_**22-590780369821e-20*_**24+209115148595e-20*_**26+17815649583e-20*_**28-1616407246e-20*_**30-238069625e-20*_**32+5398265e-20*_**34+1975014e-20*_**36+23333e-20*_**38-11188e-20*_**40-416e-20*_**42+44e-20*_**44+3e-20*_**46:3==$?-.0013397160907194568*_**1+.003744215136379394*_**3-.0013303178919321468*_**5-.0022654660765471786*_**7+9548499998506731e-19*_**9+6010038458963604e-19*_**11-10128858286776622e-20*_**13-6865733449299826e-20*_**15+59853667915386e-20*_**17+333165985123995e-20*_**19+21919289102435e-20*_**21-7890884245681e-20*_**23-94146850813e-19*_**25+95701162109e-20*_**27+18763137453e-20*_**29-443783768e-20*_**31-224267385e-20*_**33-3627687e-20*_**35+1763981e-20*_**37+79608e-20*_**39-942e-19*_**41-713e-20*_**43+33e-20*_**45+4e-20*_**47:46483389361763383e-20*_**0-.001005660736534047*_**2+24044856573725794e-20*_**4+.0010283086149702322*_**6-7657861071755644e-19*_**8-20365286803084818e-20*_**10+2321229049106873e-19*_**12+326021442438652e-19*_**14-2557906251794953e-20*_**16-410746443891574e-20*_**18+117811136403713e-20*_**20+24456561422485e-20*_**22-2391582476734e-20*_**24-750521420704e-20*_**26+13312279416e-20*_**28+13440626754e-20*_**30+351377004e-20*_**32-151915445e-20*_**34-8915418e-20*_**36+1119589e-20*_**38+10516e-19*_**40-5179e-20*_**42-807e-20*_**44+11e-20*_**46+4e-20*_**48}

function theta(t){
    twopi = 2 * math.pi
    return (t/2.0 * math.log(t/twopi) - t/2.0 - math.pi/8.0
            + 1.0/(48.0*t) + 7.0/(5760*t**3) + 31/(80640*t**5) + 
            127/(430080*t**7) + 511/(1216512*t**9) + 1414477/(1476034560*t**11) + 
            8191/(2555904*t**13) + 118518239/(8021606400*t**15) + 5749691557/(64012419072*t**17) + 
            91546277357/(131491430400*t**19) + 23273283019/(3472883712*t**21))
}
//Riemann Z(t) formula
function RiemannZ(t, r){
    let tDivTwoPi = t / (2 * math.pi)
    let val = math.sqrt(tDivTwoPi)
    let N = math.floor(math.abs(val))
    let SUM = 0

    let i = 1
    while (i<=N){
        SUM += (math.cos(theta(t) - t * math.log(i))) / math.sqrt(i)
        i += 1
    }
    SUM *= 2
    let frac = val - N
    let k = 0
    let R = 0

    //Remainder Calculation
    while (k <= r){
        R += C(k, frac) * tDivTwoPi**(k * (-0.5))
        k += 1
    }
    let remainder = (-1)**(N-1) * tDivTwoPi**(-0.25) * R
    return SUM + remainder
}
function benchmark(s, n){
	let avg = 0
	for (let i=0;i<n;i++){
		let now = performance.now()
		zeta(s, 0.0000001)
		avg += performance.now() - now
	}
	return avg/n
}

//console.log(zeta(math.complex(1/2, 874)))
console.log(benchmark(math.complex(1, 167.235364), 100)) //Normal zeta function testing

console.log(benchmark(math.complex(1/2, 167.235364), 100)) //Riemann Siegel Critical zeta function testing (Triggered if Re(s) is 1/2). Much Much faster
```

**Comment by josdejong** [comment]
Thanks for sharing @Bobingstern . The large array with coefficients takes a couple of kilobytes, which can be an downside when using the library in the browser. Is this really worth it in your opinion? I understand this will trade kilobytes for runtime performance.

**Comment by Bobingstern** [comment]
Theres no good way I can think of to truncate the array but I think theres a slightly better method for storing it. I used sympy to compute the expressions symoblically. What I could do instead is store only the coefficient of the Chebyshev POLYNOMIAL rather than have sympy expand the entire thing into one fat polynomial. Unforutnately, I will need to figure out a fast and efficient way to compute high derivatives of the chebyshev polynomail buts I could just use the trig definitions of them to make it easier

**Comment by Bobingstern** [comment]
Theres no good way I can think of to truncate the array but I think theres a slightly better method for storing it. I used sympy to compute the expressions symoblically. What I could do instead is store only the coefficient of the Chebyshev POLYNOMIAL rather than have sympy expand the entire thing into one fat polynomial. Unforutnately, I will need to figure out a fast and efficient way to compute high derivatives of the chebyshev polynomail buts I could just use the trig definitions of them to make it easier

**Comment by Bobingstern** [comment]
They are important in the formulation of remainder terms in the Riemann-Siegel formula. The remainders are like this
$C_0=\Psi(p)$
$C_1=-\frac{1}{96\pi^2}\Psi^{(3)}(p)$
$C_2 = \frac{1}{18432\pi^4}\Psi^{(6)}(p)+\frac{1}{64\pi^2}\Psi^{(2)}(p)$
where
$\Psi(p)=\frac{\cos(2\pi(p^2-p-1/16))}{\cos(2\pi p)}$
Its clear that the high derivatives of $\Psi(p)$ are very problematic. I used a Chebyshev approximation of this function for each remainder term $C_n$. This is why the array is so big. The remainder terms are very important as they added around 5-6 digits of accuracy and even more for larger values of t.

**Comment by josdejong** [comment]
👍 thanks for the clarification.

**Comment by josdejong** [comment]
Function `zeta` is published now in `v11.10.0`, thanks again Anik.
